### PR TITLE
feat: add support for linking config analysis as evidence to incident

### LINKED ISSUE #651: Attach analysis as evidence

### PR DESCRIPTION
**Pull request by Prashanth-Nelli**
Closes #651

### CODE REVIEW

**Comment by moshloop** [comment]
+ We should create an AnalysisLink component that covers the icon / name / age and popup and then reuse that that for the evidence body
+ The modal title should be icon and analysis name

**Comment by moshloop** [comment]
+ apply the same approach for changes
+ In the popup don't use age, use date and the standard timestamp format

**Comment by Prashanth-Nelli** [comment]
@moshloop made changes as suggested

**Comment by moshloop** [comment]
@Prashanth-Nelli - The dialog title should be ec2-no-public-ip, not security which is indicated in the icon

File: src/components/ConfigAnalysisLink/ConfigAnalysisLink.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        }}
+        size="large"
+        bodyClass=""
+      >
+        {configAnalysis?.id && (
+          <>
+            <div className="flex flex-col px-4 py-4 text-gray-500 space-y-2">
+              <div className="flex flex-col space-y-6">
+                <div className="grid grid-cols-1 gap-x-4 gap-y-8 sm:grid-cols-4">
+                  <div className="sm:col-span-1">
```
Comment (moshloop) [inline src/components/ConfigAnalysisLink/ConfigAnalysisLink.tsx:57]: This should be refactored into DescriptionCard with labelStyle="top", there shouldn't be any styling at this level it should all be in DescriptionCard - this ensures that all descriptions are styled consistently throughout - e.g. Config analysis and changes descriptions now have different fonts

**Comment by moshloop** [comment]
- Don't show null details / changes
- The resource attached to change / analysis should be prefixed in the title (and be a link)
e.g. {AWS Icon} ACC123 / {security icon} vpc-flow-logs-enabled
- The type should have the type icon, not a generic diff icon
- Don't use NA, just have it blank
- It appears to show the current date, The dates (and labels) for a config analysis should be First Observed and Last Observed
- The "identified @ / changes made" should be removed in evidence and definition of done display - The dates for these should only display in table format for the Changes / Analysis sidebars. 
- The config change analysis evidence should should just be {resource icon} {resource name} / {change icon} {change name}
- The link component should never render anything more than a link (to a config) or to a popup, the date should be the loop i.e. don't remove the table in ConfigInsights / Analysis

File: src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx [hunk 2]
```diff
@@ -39,85 +76,136 @@ export function ConfigDetailsChanges({
...
+        open={open}
+        onClose={(e) => {
+          e?.stopPropagation();
+          setOpen(false);
+        }}
+        size="large"
+        bodyClass=""
+      >
+        <div
+          className="flex flex-col h-full"
```
Comment (moshloop) [inline src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx:105]: Why do we need this styling? should't it be part of the modal ?

File: src/components/DescriptionCard/index.tsx [hunk 3]
```diff
@@ -1,28 +1,56 @@
-type DescriptionCardProps = {
+import clsx from "clsx";
+
+type DescriptionCardProps = React.HTMLProps<HTMLDivElement> & {
   items: {
-    label?: string;
+    label?: string | React.ReactNode;
     value: React.ReactNode;
   }[];
+  noOfCols?: number;
```
Comment (moshloop) [inline src/components/DescriptionCard/index.tsx:8]:
```suggestion
  columns?: number;
```

File: src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx [hunk 4]
```diff
@@ -39,85 +76,136 @@ export function ConfigDetailsChanges({
...
+            <Icon
+              name={changeDetails?.change_type}
+              secondary="diff"
+              className="w-5 h-auto pr-1"
+            />
+            {changeDetails?.change_type}
+          </>
+        }
+        open={open}
+        onClose={(e) => {
```
Comment (moshloop) [inline src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx:97]: Why do we need this ?

**Review by moshloop — CHANGES_REQUESTED** [review]

File: src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx (hunk 2 above)
Comment (Prashanth-Nelli) [reply src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx:105]: child elements layout can column wise or row wise and it depends what kind of layout we want to show inside Modal. so we can't just move this into Modal. 

I think we can assign these classes to `bodyClass` prop of `Modal`.

File: src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx (hunk 4 above)
Comment (Prashanth-Nelli) [reply src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx:97]: When a Modal is opened on top of another Modal close is not working properly. In order to fix that issue I have added this.

Comment (moshloop) [reply src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx:97]: But then shouldn't it be on all modals ? and/or add a comment on why it is needed

Comment (Prashanth-Nelli) [reply src/components/ConfigDetailsChanges/ConfigDetailsChanges.tsx:97]: I'll add a comment

**Comment by moshloop** [comment]
@Prashanth-Nelli 
- The config name and icon should not be shown in the config page - it is already in the title.
- Elsewhere it is missing the config_icon

**Pull request merged**